src/commands/stepfunctions/helpers.ts

```diff
@@ -79,3 +79,80 @@ export const parseArn = (
 export const buildLogAccessPolicyName = (stepFunction: DescribeStateMachineCommandOutput): string => {
   return `LogsDeliveryAccessPolicy-${stepFunction.name}`
 }
+
+export const injectContextIntoLambdaPayload = async (
+  describeStateMachineCommandOutput: DescribeStateMachineCommandOutput,
+  stepFunctionsClient: SFNClient,
+  context: BaseContext,
+  dryRun: boolean
+): Promise<void> => {
+  if (typeof describeStateMachineCommandOutput.definition !== 'string') {
+    return
+  }
+  let definitionHasBeenUpdated = false
+  const definitionObj = JSON.parse(describeStateMachineCommandOutput.definition) as StateMachineDefinitionType
+  for (const stepName in definitionObj.States) {
+    if (definitionObj.States.hasOwnProperty(stepName)) {
+      const step = definitionObj.States[stepName]
+      if (shouldUpdateStepForTracesMerging(step)) {
+        updateStepObject(step)
+        definitionHasBeenUpdated = true
+      }
+    }
+  }
+  if (definitionHasBeenUpdated) {
+    await updateStateMachineDefinition(
+      stepFunctionsClient,
+      describeStateMachineCommandOutput,
+      definitionObj,
+      context,
+      dryRun
+    )
+  }
+}
+
+export const updateStepObject = ({Parameters}: StepType): void => {
+  if (Parameters) {
+    Parameters[`Payload.$`] = 'States.JsonMerge($$, $, false)'
+  }
+}
+
+export const shouldUpdateStepForTracesMerging = (step: StepType): boolean => {
+  // is default lambda api
+  if (step.Resource === 'arn:aws:states:::lambda:invoke') {
+    if (step.Parameters === undefined) {
```

Comment:
nit,  if (step.Parameters) is expected to be better. Check [this](https://stackoverflow.com/questions/28975896/is-there-a-way-to-check-for-both-null-and-undefined#:~:text=450-,if(%20value%20)%20%7B%0A%7D,-will%20evaluate%20to)

Reply:
Thanks for the comment. Updated.